### ISSUE TITLE
There's no way to trigger a specific log message to be closed.

### ISSUE THREAD

**Issue by simonexmachina** [fabien-d/alertify.js#89]
Unless I'm missing something...?

**Comment by fabien-d** [comment]
From code, no. You can click to close, plus it automatically closes after set time. Can you provide a use case for this?

**Comment by simonexmachina** [comment]
This is related to my [other issue](https://github.com/fabien-d/alertify.js/issues/88#issuecomment-12180024). My use case is this:
1. User clicks a button that triggers a process that can take a number of seconds.
2. The system shows a notification that the operation is in progress
3. When the operation is completed the first notification is closed and another notification is displayed to indicate the outcome

There's two things I'd like to do here that I can't using the current API:
1. Use Spin.js to display a progress indicator, but this requires a DOM element to attach itself to and I can't get this from the Alertify API
2. Close the first notification when the operation completes

**Comment by fabien-d** [comment]
I see your point. I'm thinking I may return the log message instead of the alertify object... This would take away chaining on log messages, but would return the newly created log message.

I'll have to consider a few options, see what makes the most sense.

**Comment by simonexmachina** [comment]
My suggestion would be to support chaining through the `Alertify` global, which internally uses an OO `Alertify` prototype. Then for this scenario API users could use `Aleritfy.create()` or `new Alertify()` then have access to instance-specific operations. This Alertify prototyp coule expose the DOM element through a method or a property (ie. .element(), .el, .getElement() etc.) and have a .close() method.

```
function() {
  Alertify = function() {...}
  Alertify.prototype = {
    el: null,
    alert: function() {
      ...
      return global;
    }, 
    close: function() {...}
  }
  return global = {
    alert: function() {
      return new Alertify().alert();
    },
    ...
    create: function() {
      return new Alertify();
    }
  }
}

Alertify.alert('Hello world')
var obj = Alertify.create();
obj.alert('Hello world')
new Spinner().spin(obj.el)
obj.close();  
```

**Comment by 235** [comment]
I'm facing the same issue. I have to show a permanent alert, something like `alertify.log( "Make some action to progress!", "custom",0 )`, until a user will take some actions. Then my code has to close it.

**Comment by fabien-d** [comment]
@simonwade this is a good approach. I was thinking of separating the log message and the dialog message into [separate modules](https://github.com/fabien-d/alertify.js/issues/90). I would keep the dialog module as a singleton and restructure the log module as a factory.

The idea would be to offer a custom build option for alertify to only include the parts needed.

**Comment by simonexmachina** [comment]
I'm not against code separation, but the build process should be optional IMO. One of the things I like about Alertify was that I could bring it in using bower and it worked straight away. So I think that adding the .create() method (or whatever it's called) and, if you do decide to modularise, then include a full build including both dialogs and log messages in the repo.

**Comment by fabien-d** [comment]
The custom build would be optional - similar to [Modernizr](http://modernizr.com/download/).

Bower would still pull in the full core library (no extension). The custom build option would simply be available for extensions that shouldn't be in the core library, but could be useful to some.

**Comment by simonexmachina** [comment]
Okay, sounds great!

Cheers,

Simon

**Comment by fabien-d** [comment]
This issue is resolved in `0.4.0rc1`. You can now do:

``` js
var log = Alertify.log.info("Notification Message", 0);

// close
log.close();

// after closing, you can recreate and reshow
log.create();
log.show();

// you can also get the log element
// returns DOM element
log.el
```

You can try it out on the [demo site](http://fabien-d.github.com/alertify.js/0.4.0rc1/)
